packages/snap-preact-components/src/components/Molecules/Rating/Rating.tsx

```diff
@@ -6,84 +6,155 @@ import { jsx, css } from '@emotion/react';
+import { defined, mergeProps } from '../../../utilities';
+import { Icon, IconProps } from '../../Atoms/Icon';
 
 const CSS = {
-	rating: (emptyRatingSrc?: string, fullRatingSrc?: string) =>
+	rating: ({}: Partial<RatingProps>) =>
 		css({
-			textAlign: 'left',
-			height: '24px',
-			margin: '10px auto',
-
-			'& .emptyRatingBox': {
-				width: '129px',
-				height: '24px',
-				float: 'left',
-				backgroundRepeat: 'no-repeat',
-				background: `url(${emptyRatingSrc || '//4tcdn.blob.core.windows.net/4tjs1/images/allwallstarsempty.png'}) no-repeat`,
-				textAlign: 'center',
-				border: '0px !important',
+			display: 'flex',
+			alignItems: 'center',
+
+			'& .ss__rating__icons': {
+				position: 'relative',
+			},
+
+			'& .ss__rating__stars': {
+				width: '100%',
+				display: 'inline-grid',
+				gridTemplateColumns: '20% 20% 20% 20% 20%',
+
+				'&.ss__rating__stars--full': {
+					position: 'absolute',
+					top: 0,
+					left: 0,
+				},
+
+				'&.ss__rating__stars--empty': {
+					position: 'relative',
+				},
 			},
 
-			'& .fullRatings': {
-				background: `url(${fullRatingSrc || '//4tcdn.blob.core.windows.net/4tjs1/images/allwallstarsfull.png'}) no-repeat`,
-				height: '24px',
-				textAlign: 'left',
-				backgroundRepeat: 'no-repeat',
-				border: '0px !important',
-				float: 'left',
+			'& .ss__rating__stars__star': {
+				overflow: 'hidden',
 			},
 		}),
 };
 
 export const Rating = observer((properties: RatingProps): JSX.Element => {
 	const globalTheme: Theme = useTheme();
 
-	const defaultProps: Partial<RatingProps> = {};
+	const defaultProps: Partial<RatingProps> = {
+		fullIcon: 'star',
+		emptyIcon: 'star-o',
+	};
 
 	const props = mergeProps('rating', globalTheme, defaultProps, properties);
 
-	const { emptyRatingSrc, fullRatingSrc, showEmptyRatings, count, disableStyles, className, style } = props;
+	const { alwaysRender, count, text, disablePartialFill, emptyIcon, fullIcon, disableStyles, className, style, styleScript } = props;
 
-	let rating = props.rating;
+	const subProps: RatingSubProps = {
+		fullIcon: {
+			// default props
+			icon: fullIcon,
+			// global theme
+			...globalTheme?.components?.icon,
+			// inherited props
+			...defined({
+				disableStyles,
+			}),
+			// component theme overrides
+			theme: props?.theme,
+		},
+		emptyIcon: {
+			// default props
+			icon: emptyIcon,
+			color: '#ccc',
+			// global theme
+			...globalTheme?.components?.icon,
+			// inherited props
+			...defined({
+				disableStyles,
+			}),
+			// component theme overrides
+			theme: props?.theme,
+		},
+	};
+
+	// normalize values to ensure good rendering
+	let value = props.value;
+	if (isNaN(value)) {
+		value = Number(value) || 0;
+	}
+	if (value < 0) {
+		value = 0;
+	}
+	if (value > 5) {
+		value = 5;
+	}
 
 	const styling: { css?: StylingCSS } = {};
-	if (!disableStyles) {
-		styling.css = [CSS.rating(emptyRatingSrc, fullRatingSrc), style];
+	const stylingProps = { ...props };
```

Comment:
Should `value` be added here so that it can be accessed in the styleScript and so that it overwrites `props.value` since it's a primitive and would looses reference if the above if statements change it's value?

Reply:
Ah yes, I will update stylingProps accordingly.